ocrd_utils/ocrd_utils/str.py

```diff
@@ -62,25 +62,29 @@ def make_file_id(ocrd_file, output_file_grp):
     Derive a new file ID for an output file from an existing input file ``ocrd_file``
     and the name of the output file's ``fileGrp/@USE``, ``output_file_grp``.
     If ``ocrd_file``'s ID contains the input file's fileGrp name, then replace it by ``output_file_grp``.
-    Else if ``ocrd_file``'s ID contains the input file's pageId, then merely append ``output_file_grp``.
-    Otherwise use ``output_file_grp`` together with the position of ``ocrd_file`` within the input fileGrp
-    (as a fallback counter), and increment counter until there is no more ID conflict.
-    """
+    Else if ``ocrd_file``'s ID contains the input file's pageId, then merely append it to ``output_file_grp``.
+    Else if ``ocrd_file`` has a pageId at all, then merely append it to ``output_file_grp``.
+    Otherwise use ``output_file_grp`` together with the position of ``ocrd_file`` within the input fileGrp.
+    """
+    # considerations for this behaviour:
+    # - uniqueness (in spite of different METS and processor conventions)
+    # - predictability (i.e. output name can be anticipated from the input name)
+    # - stability (i.e. output at least as much sorted and consistent as the input)
+    # ... and all this in spite of --page-id selection and --overwrite
     ret = ocrd_file.ID.replace(ocrd_file.fileGrp, output_file_grp)
     if ret == ocrd_file.ID:
```

Comment:
@kba, BTW we could add another feature easily here:
```suggestion
    if ret == ocrd_file.ID and ocrd_file.fileGrp != output_file_grp:
```

This change looks harmless, but it will allow processing _in-place_ on demand by simple passing the input fileGrp as the output fileGrp together with `--overwrite`. (That's something we don't support officially, but sometimes it is really helpful to avoid creating too many fileGrps, esp. during GT production.)

Reply:
Implemented